specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/assessmentsMetadata.json

```diff
@@ -0,0 +1,358 @@
+      "get": {
+        "x-ms-examples": {
+          "Get security assessment metadata": {
+            "$ref": "./examples/AssessmentsMetadata/GetAssessmentsMetadata_example.json"
+          }
+        },
+        "tags": [
+          "Assessments Metadata"
+        ],
+        "description": "Get metadata information on an assessment type",
+        "operationId": "AssessmentsMetadata_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/AssessmentsMetadataName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SecurityAssessmentMetadata"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/assessmentsMetadata": {
+      "get": {
+        "x-ms-examples": {
+          "List security assessment metadata for subscription": {
+            "$ref": "./examples/AssessmentsMetadata/ListAssessmentsMetadata_subscription_example.json"
+          }
+        },
+        "tags": [
+          "Assessments Metadata"
+        ],
+        "description": "Get metadata information on all assessment types in a specific subscription",
+        "operationId": "AssessmentsMetadataSubscription_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/SecurityAssessmentMetadataList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/assessmentsMetadata/{assessmentsMetadataName}": {
+      "get": {
+        "x-ms-examples": {
+          "Get security assessment metadata for subscription": {
+            "$ref": "./examples/AssessmentsMetadata/GetAssessmentsMetadata_subscription_example.json"
+          }
+        },
+        "tags": [
+          "Assessments Metadata"
+        ],
+        "description": "Get metadata information on an assessment type in a specific subscription",
```

Comment:
>ype in a specific subscriptio [](start = 67, length = 29)

I am unclear on the need of subscription specific apis..
How can assessment types be different for different subscriptions ?

Could you also clarify what is resource type 'assessment type 'refer to here ?
If it is /assessment/assessmentName resource type, how is these two resource types are different.. ?

Please schedule a meeting with api review board ..

Reply:
Hi,

We are adding the notion of 'custom assessments' to ASC, giving the user the ability to create custom policy in Azure Policy, and load it as assessment to ASC. 

The custom assessments are defined in subscription scope, thus different subscriptions can have different assessments. This is the reason why we need metadata API in subscription scope.

the 'assessmentType' field represent whether the assessment is a BuilltIn assessment of ASC or a custom assessment defined by the user. This field is equivelent to the 'policyType' field in Azure Policy policy definitions API.

Thanks,
Tal.